test/unit/tedious-test.js

```diff
@@ -23,3 +23,27 @@ exports.connection = function(test) {
 
   test.done();
 };
+
+exports.connectionDoesNotModifyPassedConfig = function(test) {
+  var config = {
+    userName: 'sa',
+    password: 'sapwd',
+    options: {
+      port: 1234,
+      cryptoCredentialsDetails: {
+        ciphers: 'RC4-MD5'
+      }
+    }
+  };
+
+  var connection = new Connection(config);
+
+  test.notStrictEqual(connection.config, config);
+  test.notStrictEqual(connection.config.options, config.options);
+
+  // Test that we did not do a deep copy of the cryptoCredentialsDetails,
+  // as we never modify that value inside tedious.
+  test.strictEqual(connection.config.options.cryptoCredentialsDetails, config.options.cryptoCredentialsDetails);
```

Comment:
Ah, yes, sorry, I missed this.

`cryptoCredentialsDetails` is not a tedious specific option, it's just passed forwards to https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options. I don't think we need to care about what happens to this object, and if it get's modified by Node.js itself, that is a bug in Node.js that should be fixed there.